File: Data Structures & Algorithms/is-palindrome/submission-3.py
import string
class Solution:
    def isPalindrome(self, s: str) -> bool:
        if len(s)<=1:
            return True
        else:
            new_str = s.translate(str.maketrans('', '', string.punctuation)).replace(" ", "").lower()
            print(new_str)
            left = 0
            right = len(new_str) - 1
            while left < right:
                if new_str[left] == new_str[right]:
                    left += 1
                    right -= 1
                else:
                    return False
            return True

File: Data Structures & Algorithms/generate-parentheses/submission-3.cpp
class Solution {
public:
    vector<string> generateParenthesis(int n) {
        vector<vector<string>> tab(n+1);
        tab[0] = {""};

        for(int i = 0; i <= n; i++)
        {
            for(int j = 0; j < i; j++)
            {
                for(const string& left: tab[j])
                {
                    for(const string& right: tab[i-j-1])
                    {
                        tab[i].push_back("(" + left + ")" + right);
                    }
                }
            }
        }

        return tab[n];
    }
};
